cfme/cloud/instance/__init__.py

```diff
@@ -376,6 +380,125 @@ def unset_ownership(self):
         view.flash.assert_no_error()
         view.flash.assert_success_message(msg)
 
+    def set_retirement_date(self, when=None, offset=None, warn=None):
+        """Overriding common method to use widgetastic views/widgets properly
+
+        Args:
+            when: When to retire.
+                :py:class:`datetime.datetime` or :py:class:`utils.timeutil.parsetime`.
+            offset: :py:class:`dict` with months, weeks, days, hours keys. other keys ignored
+            warn: When to warn, fills the select in the form in case the ``when`` is specified.
+
+        Note: this should be moved up to the common VM class when infra+cloud+common are all WT
+
+        If when and offset are both None, this removes retirement date
+
+        Offset is dict to remove ambiguity between timedelta/datetime and months/weeks/days/hours
+        timedelta supports creation with weeks, but not months
+        timedelta supports days attr, but not weeks or months
+        timedelta days attr will report a total summary, not the component that was passed to it
+        For these reasons timedelta isn't appropriate for offset
+        An enhancement to cfme.utils.timeutil extending timedelta would be great for making this a
+        bit cleaner
```

Comment:
Maybe a sample call of this function could be added to docscrting to make it more clear

Reply:
Good call, will do. These notes are mostly for saving others time if they start to consider making offset a timedelta with months. An example would be very useful to callers.